pkg/apiserver/handlers.go

```diff
@@ -139,6 +139,13 @@ func tooManyRequests(req *http.Request, w http.ResponseWriter) {
 func RecoverPanics(handler http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
 		defer runtime.HandleCrash(func(err interface{}) {
+			if e, ok := err.(error); ok {
```

Comment:
I was thinking you'd just not wrap the timeout handler with `RecoverPanics`. Instead, wrap the handler you pass to timeout handler.